### PR TITLE
Fix mean calculation

### PR DESCRIPTION
**Pull request by Mupico**
*Issue #, if available:*

*Description of changes:* The mean was calculated in a wrong way when filtering the result set with a where-clause. Fixed that calculation and wrote a test for it. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by tdhd** [comment]
Thank you very much for fixing this!

File: src/test/scala/com/amazon/deequ/analyzers/AnalyzerTests.scala [hunk 1]
```diff
@@ -418,6 +418,11 @@ class AnalyzerTests extends WordSpec with Matchers with SparkContextSpec with Fi
       val df = getDfFull(sparkSession)
       assert(Mean("att1").calculate(df).value.isFailure)
     }
+    "compute mean correctly for numeric data with filtering" in withSparkSession { sparkSession =>
```
Comment (tdhd) [inline src/test/scala/com/amazon/deequ/analyzers/AnalyzerTests.scala]:
```suggestion
    "compute mean correctly for numeric data with where predicate" in withSparkSession { sparkSession =>
```

**Comment by sscdotopen** [comment]
Great catch, thank you!

**Comment by tdhd** [comment]
My test rename suggestion violated our style checks ;)

```error file=/home/travis/build/awslabs/deequ/src/test/scala/com/amazon/deequ/analyzers/AnalyzerTests.scala message=File line length exceeds 100 characters line=421```

**Pull request merged**